src/Microsoft.Extensions.ServiceDiscovery/Http/HttpServiceEndPointResolver.cs

```diff
@@ -159,7 +159,7 @@ private sealed class ResolverEntry : IAsyncDisposable
     {
         private readonly ServiceEndPointResolver _resolver;
         private readonly IServiceEndPointSelector _selector;
-        private const ulong CountMask = unchecked((ulong)-1);
+        private const ulong CountMask = ~(RecentUseFlag | DisposingFlag);   
         private const ulong RecentUseFlag = 1UL << 61;
         private const ulong DisposingFlag = 1UL << 62;
```

Comment:
Is there a reason we are leaving the top bit unused? Why aren't these:

```C#
        private const ulong RecentUseFlag = 1UL << 62;
        private const ulong DisposingFlag = 1UL << 63;
```

Reply:
Maybe there was at some point, but not any more